common/namespace/handler.go

```diff
@@ -787,3 +781,65 @@ func (d *HandlerImpl) validateVisibilityArchivalURI(URIString string) error {
 
 	return archiver.ValidateURI(URI)
 }
+
+// validateRetentionDuration ensures that retention duration can't be set below a sane minimum.
+func validateRetentionDuration(retention time.Duration, isGlobalNamespace bool) error {
+	min := MinRetentionLocal
+	max := common.MaxWorkflowRetentionPeriod
+	if isGlobalNamespace {
+		min = MinRetentionGlobal
+	}
+	if retention < min || retention > max {
+		return errInvalidRetentionPeriod
+	}
+	return nil
+}
+
+func validateStateUpdate(oldNamespace *persistence.GetNamespaceResponse, newNamespace *workflowservice.UpdateNamespaceRequest) error {
+	if newNamespace.UpdateInfo == nil {
```

Comment:
I find your choice of local variable names confusing here.  Isn't the old namespace guaranteed to be identical to the new namespace?   It's not that I didn't figure out what you're really doing, I just think it's obscured by the names. Maybe:

validateStateUpdate(existingNsDetails *persistence.GetNamespaceResponse, nsUpdateRequest *workflowservice.UpdateNamespaceRequest) error